### PR TITLE
package etesync-dav

### PR DESCRIPTION
**Pull request by Valodim**
###### Motivation for this change

Packaged the etesync-dav component of [etesync](https://www.etesync.com/) :)

This PR includes python library dependencies pyscrypt, orderedmultidict, and furl.

The furl package had a failing unit test, which I filed a bug report for: https://github.com/gruns/furl/issues/121

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @tasn (etesync), @ricmoo (pyscrypt), @gruns (orderedmultidict, furl)

### CODE REVIEW

File: maintainers/maintainer-list.nix
Comment (jonringer) [inline maintainers/maintainer-list.nix]: this should be a separate commit:

```
maintainers: add valodim
```

File: pkgs/applications/misc/etesync-dav/default.nix
Comment (jonringer) [inline pkgs/applications/misc/etesync-dav/default.nix]: personally, i would refactor this package into a `etesync.nix`, and just import it.

File: pkgs/development/python-modules/furl/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/furl/default.nix]:
```suggestion
  checkPhase = ''
    pytest -k 'not join'
  '';
```

File: pkgs/development/python-modules/furl/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/furl/default.nix]:
```suggestion
    flake8
    pytest
```

File: pkgs/development/python-modules/pyscrypt/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/pyscrypt/default.nix]:
```suggestion
    ${python.interpreter} tests/run-tests-hash.py
```

File: pkgs/development/python-modules/pyscrypt/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/pyscrypt/default.nix]:
```suggestion
{ lib, buildPythonPackage, fetchPypi, python }:
```

File: pkgs/applications/misc/etesync-dav/default.nix
Comment (Valodim) [reply pkgs/applications/misc/etesync-dav/default.nix]: Hi Jon, thanks for the quick review! I added all the other changes. I'm not sure how to do this one, though. Do you mean package it as a separate package, i.e. `etesync/default.nix`? Or actually have an `etesync.nix` in this one? If so, I'm not quite sure how to do that and couldn't find an example in another package to copy from.

**Comment by gruns** [comment]
@Valodim Weighing in here: is there anything in furl, eg the test that mysteriously fails for you (https://github.com/gruns/furl/issues/121), that blocks this PR?

**Comment by jonringer** [comment]
@gruns nix builds these packages in a sandboxed environment (chroot + no network + no FHS), so there's a large possibility that some intended test behavior may be invalidated. Doesn't mean it can't be worked around.

**Comment by Valodim** [comment]
the failing test doesn't look like it would be caused by sandboxing to me. it concerns edge case behavior of furl to mirror that of urllib. upstream should probably figure out what's going on, but from nixpkg's point of view I guess it's not terribly important to have it fixed before we package it.

File: pkgs/applications/misc/etesync-dav/default.nix
Comment (Valodim) [reply pkgs/applications/misc/etesync-dav/default.nix]: I just pulled it out into its own package :)

**Comment by Valodim** [comment]
I think this is good to merge now. I've been using it for a couple days now, works as intended for me.

The one failing test in furl doesn't seem like a dealbreaker to me (it tests for consistency with urllib behavior in edge cases), and if upstream ever figures out what's going on we can enable it again :)

File: pkgs/development/python-modules/etesync/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/etesync/default.nix]: please add meta section

https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md

File: pkgs/development/python-modules/orderedmultidict/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/orderedmultidict/default.nix]: added yourself as maintainer

File: pkgs/development/python-modules/pyscrypt/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/pyscrypt/default.nix]: add yourself as maintainer

File: pkgs/development/python-modules/etesync/default.nix
Comment (Valodim) [reply pkgs/development/python-modules/etesync/default.nix]: whoops. done, sorry about that.

I feel kind of bad causing human roundtrips on stuff like this :hurtrealbad: but also man there is a lot of stuff to remember. Can't stuff like this be encoded in CI roundrips and admonished by bots? Anyways, thanks for your work!

File: pkgs/applications/misc/etesync-dav/default.nix
Comment (matt-snider) [inline pkgs/applications/misc/etesync-dav/default.nix]: I'm pretty new to NixOS, but had been packaging this myself until I came across your PR. Looks good! :+1:

It seems like `callPackage` isn't used here
 
```suggestion
{ lib, python3Packages, radicale2 }:
```

**Comment by matt-snider** [comment]
@Valodim There's been some minor releases. For etesync/-dav it might be good to bump them and see if everything still works properly

**Comment by jonringer** [comment]
also,o please resolve conflicts

**Comment by tasn** [comment]
Hey, maintainer of etesync here.
I just noticed you are packaging "pyscrypt". That's fine, but just fyi, the package scrypt is faster (uses opensssl), so it's better to use that if possible. I'll take a look at changing the pypi package to depend on that by default and have pyscript as an alternative to avoid such confusing in the future.

Looking forward to seeing etesync-dav packaged.

File: pkgs/applications/misc/etesync-dav/default.nix
Comment (Valodim) [reply pkgs/applications/misc/etesync-dav/default.nix]: thanks!

**Comment by Valodim** [comment]
done and done.

File: pkgs/development/python-modules/etesync/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/etesync/default.nix]:
```suggestion
  version = "0.9.2";
  disabled = isPy27;
```

File: pkgs/development/python-modules/etesync/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/etesync/default.nix]:
```suggestion
{ lib, buildPythonPackage, fetchPypi, isPy27
```

**Review by jonringer — COMMENTED** [review]
to avoid:
```
==================================== ERRORS ====================================
__________________ ERROR collecting tests/test_collections.py __________________
/nix/store/7lx9rx7m8sc9wpjqkng5r4c3fy2mn3bv-python2.7-pytest-4.6.8/lib/python2.7/site-packages/_pytest/python.py:507: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
/nix/store/905b3mx703j6an71ydz9s0lwi0c1a70w-python2.7-py-1.8.1/lib/python2.7/site-packages/py/_path/local.py:701: in pyimport
    __import__(modname)
E     File "/build/etesync-0.9.2/tests/test_collections.py", line 144
E   SyntaxError: Non-ASCII character '\xd7' in file /build/etesync-0.9.2/tests/test_collections.py on line 144, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details
!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
```
I guess it technically is erroring on tests, and might be python2 compatible. But likely not

**Comment by Valodim** [comment]
squashed that in :+1:

**Review by jonringer — COMMENTED** [review]
commit history should be:
```
pythonPackages.pyscrypt: init at 1.6.2
pythonPackages.furl: 2.0.0 -> 2.1.0
pythonPackages.etesync: init at 0.9.2
etesync-dav: init at 0.14.0 
```

**Comment by jonringer** [comment]
i would recommend using `git rebase -i HEAD~4` and using the `reword` command

**Comment by Valodim** [comment]
Done. Fixed version numbers, too.

**Comment by jonringer** [comment]
> Hey, maintainer of etesync here.
> I just noticed you are packaging "pyscrypt". That's fine, but just fyi, the package scrypt is faster (uses opensssl), so it's better to use that if possible. I'll take a look at changing the pypi package to depend on that by default and have pyscript as an alternative to avoid such confusing in the future.
> 
> Looking forward to seeing etesync-dav packaged.

@tasn thanks for the heads up! 

I'm not a user of this package, so I'll defer to @Valodim as to what should be in it. But we can always improve the packaging later. I'm mostly concerned about usability and correctness.

**Review by jonringer — APPROVED** [review]
diff LGTM
commits LGTM

```
[14 built, 11 copied (5.4 MiB), 1.3 MiB DL]
https://github.com/NixOS/nixpkgs/pull/76886
9 package built:
etesync-dav python27Packages.furl python27Packages.pyscrypt python37Packages.etesync python37Packages.furl python37Packages.pyscrypt python38Packages.etesync python38Packages.furl python38Packages.pyscrypt
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build etesync-dav python27Packages.furl python27Packages.pyscrypt python37Packages.etesync python37Packages.furl python37Packages.pyscrypt python38Packages.etesync python38Packages.furl python38Packages.pyscrypt

**Comment by jonringer** [comment]
darwin failures don't seem to be related to this PR

**Comment by tasn** [comment]
> > Hey, maintainer of etesync here.
> > I just noticed you are packaging "pyscrypt". That's fine, but just fyi, the package scrypt is faster (uses opensssl), so it's better to use that if possible. I'll take a look at changing the pypi package to depend on that by default and have pyscript as an alternative to avoid such confusing in the future.
> > Looking forward to seeing etesync-dav packaged.
> 
> @tasn thanks for the heads up!
> 
> I'm not a user of this package, so I'll defer to @Valodim as to what should be in it. But we can always improve the packaging later. I'm mostly concerned about usability and correctness.

Fair enough, it doesn't *really* matter, but it's just something I noticed. Great to see it merged. :)

**Pull request merged**
